### PR TITLE
feat: Flexmeasures api client setup

### PR DESCRIPTION
**Pull request by GustaafL**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -84,13 +88,13 @@ testing =
...
 # CAUTION: --cov flags may prohibit setting breakpoints while debugging.
 #          Comment those flags to avoid this pytest issue.
-addopts =
-    --cov flexmeasures_client --cov-report term-missing
-    --verbose
-norecursedirs =
-    dist
-    build
-    .tox
+# addopts =
```
Comment (nhoening) [inline setup.cfg:91]: Why is this commented out?

File: src/flexmeasures_client/client.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+class FlexmeasuresClient:
+    """Main class for connecting to the Flexmeasures API"""
+
+    password: str
+    email: str
+    access_token: str = None
+    scheme: str = "http"
+    host: str = "localhost:5000"
+    local: bool = False
+    version: str = "/v3_0/"
```
Comment (nhoening) [inline src/flexmeasures_client/client.py:24]: Should say `api_version`, and including the prefixed slash should not be necessary.

File: src/flexmeasures_client/client.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        method: str = "POST",
+        path: str = f"/api{version}",
+        params: dict[str, Any] | None = None,
+        json: dict | None = None,
+        headers: dict | None = None,
+    ) -> tuple[dict, int]:
+        url = URL.build(scheme="http", host="localhost:5000", path=path).join(
+            URL(uri),
+        )
+        print(url)
```
Comment (nhoening) [inline src/flexmeasures_client/client.py:45]: This debug statement should go out or become a real logging entry.

File: src/flexmeasures_client/client.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        self,
+        uri: str,
+        *,
+        method: str = "POST",
+        path: str = f"/api{version}",
+        params: dict[str, Any] | None = None,
+        json: dict | None = None,
+        headers: dict | None = None,
+    ) -> tuple[dict, int]:
+        url = URL.build(scheme="http", host="localhost:5000", path=path).join(
```
Comment (nhoening) [inline src/flexmeasures_client/client.py:42]: host should come from `self`, or not?

File: src/flexmeasures_client/client.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    email: str
+    access_token: str = None
+    scheme: str = "http"
+    host: str = "localhost:5000"
+    local: bool = False
+    version: str = "/v3_0/"
+
+    request_timeout: float = 10.0
+    session: ClientSession | None = None
+
```
Comment (nhoening) [inline src/flexmeasures_client/client.py:28]: Maybe a dedicated `__init__` function, where one can start a client with chosen parametrisation, e.g. for `host`, and we get an error if email and password are not given?

Not sure what the best way is to store credentials like password, we could figure that out in a follow-up issue.

**Review by nhoening — COMMENTED** [review]
Interesting work, and so I decided to throw in some early comments :)

Also looking forward to some `assert` statements in tests :)

File: src/flexmeasures_client/client.py [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+            json={
+                "email": self.email,
+                "password": self.password,
+            },
+        )
+        self.access_token = response["auth_token"]
+
+    async def post_measurements(
+        self,
+        access_token: str,
```
Comment (victorgarcia98) [inline src/flexmeasures_client/client.py:99]: Here I would use `self.access_token`.

**Review by victorgarcia98 — CHANGES_REQUESTED** [review]
Good stuff! Basically, there are some exception types that I would change, just to make the exception handling more clear when we use the client.

Other than that, I've added some ideas for testing that we could implement.

**Pull request closed without merging**